lib/jbuild

```diff
@@ -0,0 +1,31 @@
+(* -*- tuareg -*- *)
+#require "unix"
+
+let flags pkgs =
+  let cmd = "ocamlfind ocamlc -verbose" ^ (
+    List.fold_left (fun acc pkg -> acc ^ " -package " ^ pkg) "" pkgs
+  ) in
+  let ic = Unix.open_process_in
+    (cmd ^ " | grep -oEe '-ppx \"[^\"]+\"'")
```

Comment:
As you pointed out, this also is not good enough, because the old version of cstruct (2.3.0) will output `-ppx ppx_cstruct` instead of `-ppx "ppx_cstruct --as-ppx"`. So we need to support escaped quoted strings and unescaped ones. The regexp becomes the horrible monster that follows: `grep -oEe '-ppx ("([^"\\]|\\.)+"|\w+)'`